File: src/com/soywiz/flash/util/Color.scala
package com.soywiz.flash.util

object Color {
  lazy val black = new Color(0, 0, 0, 255)
  lazy val red = new Color(255, 0, 0, 255)
  def apply(v:Int) = new Color(
    (v >>> 0) & 0xFF,
    (v >>> 8) & 0xFF,
    (v >>> 16) & 0xFF,
    (v >>> 24) & 0xFF
  )
}

class Color(val r:Int, val g:Int, val b:Int, val a:Int) {
  def toInt: Int = (a << 24) | (r << 16) | (b << 8) | (g << 0)
}

File: src/com/soywiz/flash/util/Color.java
package com.soywiz.flash.util;

public class Color {
    public final int r, g, b, a;

    public Color(int rgba) {
        this((rgba >> 16) & 0xFF, (rgba >> 8) & 0xFF, (rgba >> 0) & 0xFF, (rgba >> 24) & 0xFF);
    }

    public Color(int r, int g, int b, int a) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }

    public int toInt() {
        return (a << 24) | (r << 16) | (g << 8) | (b << 0);
    }

    static public Color black = new Color(0, 0, 0, 255);
    static public Color red = new Color(255, 0, 0, 255);

}
